kart/log.py

```diff
@@ -51,19 +85,62 @@
 @click.argument("args", nargs=-1, type=click.UNPROCESSED)
 def log(ctx, output_format, json_style, do_dataset_changes, with_feature_count, args):
     """ Show commit logs """
+    repo = ctx.obj.get_repo(allowed_states=KartRepoState.ALL_STATES)
+
+    # Handle positional args.
+    # These could be refs, commits, trees, or paths.
+    #  * as soon as we encounter a path, we assume all remaining args are also paths.
+    #  * i.e. the paths must be given *last*.
+    #  * If it's ambiguous whether something is a path or not, we assume it's a commit-ish.
+    #  * If you want to be unambiguous, provide the `--` arg to separate the list of commit-ish-es and paths.
+    # This behaviour is consistent with git's behaviour.
+    try:
+        dash_index = args.index("--")
+    except ValueError:
```

Comment:
Maybe it's pythonic how it is but I don't love try-except-else for readability - I kinda prefer:

try:
  dash_index = ...
except ValueError:
  dash_index = None
  
  if dash_index is None:
    ...

Reply:
I prefer the `else:` myself.

But now that I'm thinking about it, I actually prefer this to both of those:

```python

    if "--" in args:
        dash_index = args.index("--")
        paths = args[dash_index + 1 :]
        other_args = args[:dash_index]
    else:
        ...
```